sql/src/main/java/io/crate/planner/operators/LogicalPlan.java

```diff
@@ -233,4 +233,8 @@ default void execute(DependencyCarrier executor,
     }
 
     <C, R> R accept(LogicalPlanVisitor<C, R> visitor, C context);
+
+    default StatementType type() {
```

Comment:
Imho I wouldn't make it a default with impl here as it could cause troubles in the future when a new Plan is introduced for example.

Reply:
as discussed, ignore this comment. Would be nice to add the `@Override` here though.

Reply:
a LogicalPlan is always for a SELECT statement